From fcbb5ef800a9280862a57d9922a2dafcd4096fe3 Mon Sep 17 00:00:00 2001
From: Jan Pavlinec <jan.pavlinec@nic.cz>
Date: Tue, 16 Mar 2021 14:21:53 +0100
Subject: [PATCH] hostapd: enable CONFIG_NO_VLAN for all variants

This commit disables VLAN support for hostapd.

When hostapd is compiled without CONFIG_NO_VLAN option then
WPA-EAP/802.1x AP mode is not working with wifi card which does not
support NL80211_CMD_SET_STATION command and uses nl80211 hostapd
driver. hostapd is also trying to use VLANs even when they were not
enabled in the runtime config file.

If VLAN functionality is necessary it is possible to use Macvlan
subystem. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?id=79cf79abce71

Signed-off-by: Jan Pavlinec <jan.pavlinec@nic.cz>
---
 package/network/services/hostapd/files/hostapd-basic.config | 2 +-
 package/network/services/hostapd/files/hostapd-full.config  | 2 +-
 package/network/services/hostapd/files/hostapd-mini.config  | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/package/network/services/hostapd/files/hostapd-basic.config b/package/network/services/hostapd/files/hostapd-basic.config
index 461b178433..d902297bc3 100644
--- a/package/network/services/hostapd/files/hostapd-basic.config
+++ b/package/network/services/hostapd/files/hostapd-basic.config
@@ -195,7 +195,7 @@ CONFIG_NO_ACCOUNTING=y
 CONFIG_NO_RADIUS=y
 
 # Remove support for VLANs
-#CONFIG_NO_VLAN=y
+CONFIG_NO_VLAN=y
 
 # Enable support for fully dynamic VLANs. This enables hostapd to
 # automatically create bridge and VLAN interfaces if necessary.
diff --git a/package/network/services/hostapd/files/hostapd-full.config b/package/network/services/hostapd/files/hostapd-full.config
index 4fa1542767..26703fc05f 100644
--- a/package/network/services/hostapd/files/hostapd-full.config
+++ b/package/network/services/hostapd/files/hostapd-full.config
@@ -195,7 +195,7 @@ CONFIG_DEBUG_SYSLOG=y
 #CONFIG_NO_RADIUS=y
 
 # Remove support for VLANs
-#CONFIG_NO_VLAN=y
+CONFIG_NO_VLAN=y
 
 # Enable support for fully dynamic VLANs. This enables hostapd to
 # automatically create bridge and VLAN interfaces if necessary.
diff --git a/package/network/services/hostapd/files/hostapd-mini.config b/package/network/services/hostapd/files/hostapd-mini.config
index f31e6467b0..8c1c24768b 100644
--- a/package/network/services/hostapd/files/hostapd-mini.config
+++ b/package/network/services/hostapd/files/hostapd-mini.config
@@ -195,7 +195,7 @@ CONFIG_NO_ACCOUNTING=y
 CONFIG_NO_RADIUS=y
 
 # Remove support for VLANs
-#CONFIG_NO_VLAN=y
+CONFIG_NO_VLAN=y
 
 # Enable support for fully dynamic VLANs. This enables hostapd to
 # automatically create bridge and VLAN interfaces if necessary.
-- 
2.17.1

